Running just that section of tests again with this command:

```sh
just -f ./op-chain-ops/justfile test
```

Results in the test failing here:

```
=== RUN   TestMain_PreEcotoneScalar
    main_test.go:33: 
        	Error Trace:	/Users/tschnepp/Documents/repos/go/github.com/espressosystems/optimism-espresso-integration/op-chain-ops/cmd/ecotone-scalar/main_test.go:33
        	Error:      	Received unexpected error:
        	            	invalid character '#' after top-level value
        	Test:       	TestMain_PreEcotoneScalar
--- FAIL: TestMain_PreEcotoneScalar (0.72s)
=== RUN   TestMain_PostEcotoneScalar
    main_test.go:45: invalid character '#' after top-level value
--- FAIL: TestMain_PostEcotoneScalar (0.73s)
FAIL
FAIL	github.com/ethereum-optimism/optimism/op-chain-ops/cmd/ecotone-scalar	1.683s
```

I am seeing some warnings through the tests that indicate that it's loading a library with the wrong architecture:

```
ld: warning: ignoring file /tmp/libespresso_crypto_helper-x86_64-apple-darwin.a, building for macOS-arm64 but attempting to link with file built for macOS-x86_64
```

Which makes sense since the `CGO_FLAGS` that are being set are:

```
CGO_LDFLAGS='-L/tmp -lespresso_crypto_helper-x86_64-apple-darwin.a -framework Foundation -framework SystemConfiguration'
```

Yet, I'm running on an `arm64` mac.

--- a/flake.nix
+++ b/flake.nix
@@ -18,19 +18,27 @@
                   url = "https://github.com/EspressoSystems/espresso-network-go/releases/download/v0.0.34/libespresso_crypto_helper-x86_64-unknown-linux-gnu.a";
                   sha256 = "sha256:1c7ybrqjrp1709j08fk7zcr5q8hyfakvgv0m64zn2fywlqfdpszs";
                 }
-                else
+                else if system == "x86_64-darwin" then
                   pkgs.fetchurl {
                     url = "https://github.com/EspressoSystems/espresso-network-go/releases/download/v0.0.34/libespresso_crypto_helper-x86_64-apple-darwin.a";
                     sha256 = "sha256:1fbijfam49c2i2l0d56i0zgczcbh2gljc6fh63g7qq3h7b7z5wc6";
-                  };
+                  }
+                else # aarch64-darwin
+                  pkgs.fetchurl {
+                        url = "https://github.com/EspressoSystems/espresso-network-go/releases/download/v0.0.34/libespresso_crypto_helper-aarch64-apple-darwin.a";
+                        sha256 = "sha256:18iqpqm3jmvj20vdd8zz05891lw5sxqy6vhfc8ghmg55czabip2q";
+                  }
+                  ;
       cgo_ld_flags = if system == "x86_64-linux"
                       then "-L/tmp -lespresso_crypto_helper-x86_64-unknown-linux-gnu"
-                      else "-L/tmp -lespresso_crypto_helper-x86_64-apple-darwin -framework Foundation -framework SystemConfiguration"
+                      else if system == "x86_64-darwin" then "-L/tmp -lespresso_crypto_helper-x86_64-apple-darwin -framework Foundation -framework SystemConfiguration"
+                      else "-L/tmp -lespresso_crypto_helper-aarch64-apple-darwin -framework Foundation -framework SystemConfiguration" # aarch64-darwin
       ;
 
-      target_link =  if system == "x86_64-linux"
-                      then "/tmp/libespresso_crypto_helper-x86_64-unknown-linux-gnu.a"
-                      else "/tmp/libespresso_crypto_helper-x86_64-apple-darwin.a";
+      target_link =  if system == "x86_64-linux" then  "/tmp/libespresso_crypto_helper-x86_64-unknown-linux-gnu.a"
+                      else if system == "x86_64-darwin" then "/tmp/libespresso_crypto_helper-x86_64-apple-darwin.a"
+                      else "/tmp/libespresso_crypto_helper-aarch64-apple-darwin.a" # aarch64-darwin
+                      ;
 
       in
       {

